src/Connection/Factory.php

```diff
@@ -176,13 +176,17 @@ protected function prepareConnection(NodeConnectionInterface $connection)
         }
 
         if (!$connection instanceof RelayConnection) {
-            $connection->addConnectCommand(
-                new RawCommand('CLIENT', ['SETINFO', 'LIB-NAME', 'predis'])
-            );
-
-            $connection->addConnectCommand(
-                new RawCommand('CLIENT', ['SETINFO', 'LIB-VER', Client::VERSION])
-            );
+            if ($parameters->set_lib_name ?? true) {
+                $connection->addConnectCommand(
+                    new RawCommand('CLIENT', ['SETINFO', 'LIB-NAME', 'predis'])
+                );
+            }
+
+            if ($parameters->set_lib_version ?? true) {
+                $connection->addConnectCommand(
+                    new RawCommand('CLIENT', ['SETINFO', 'LIB-VER', Client::VERSION])
+                );
+            }
```

Comment:
```suggestion
            if ($parameters->set_client_info ?? true) {
                $connection->addConnectCommand(
                    new RawCommand('CLIENT', ['SETINFO', 'LIB-NAME', 'predis'])
                );
                
                $connection->addConnectCommand(
                    new RawCommand('CLIENT', ['SETINFO', 'LIB-VER', Client::VERSION])
                );
            }
```

Reply:
@vladvildanov @tillkruss Feel free to apply those changes if it makes better sense to group both. 👌

Reply:
Also I would favor `?? false` to have a sain configuration by default (to add overhead by default).